chain/chain/src/flat_storage_resharder.rs

```diff
@@ -2506,10 +2523,7 @@ mod tests {
         );
 
         // Scheduling of the shard split should fail.
-        assert_eq!(
-            sender.call_split_shard_task(),
-            FlatStorageReshardingSchedulableTaskResult::Failed
-        );
+        assert_eq!(sender.call_split_shard_task(), FlatStorageReshardingTaskResult::Failed);
         assert!(resharder.resharding_event().is_none());
         let flat_store = resharder.runtime.store().flat_store();
         assert_eq!(
```

Comment:
I might be worth to add an unit test that validates shard catchup is able to be interrupted and resumed through the postponing logic. I could look into it myself perhaps.